internal/schema/connection.go

```diff
@@ -250,7 +267,22 @@ func GetEsConnectionSchema(keyName string, isProviderConfiguration bool) *schema
 					Optional:      true,
 					Sensitive:     true,
 					DefaultFunc:   withEnvDefault("ELASTICSEARCH_API_KEY", nil),
-					ConflictsWith: []string{usernamePath, passwordPath},
+					ConflictsWith: []string{usernamePath, passwordPath, bearerTokenPath},
+				},
+				"bearer_token": {
+					Description:   "Bearer Token to use for authentication to Elasticsearch",
+					Type:          schema.TypeString,
+					Optional:      true,
+					Sensitive:     true,
+					DefaultFunc:   withEnvDefault("ELASTICSEARCH_BEARER_TOKEN", nil),
```

Comment:
These environment variables should be read in `config/elasticsearch` instead. These schema defaults won't be read as part of TF Plugin Framework resources so for now we're centralising them in the config code.